packages/tasit-identity-contract/src/GnosisSafe.test.js

```diff
@@ -0,0 +1,152 @@
+  before("", async () => {
+    ConfigLoader.setConfig(config);
+
+    provider = ProviderFactory.getProvider();
+
+    const anaPrivateKey =
+      "0x11d943d7649fbdeb146dc57bd9cfc80b086bfab2330c7b25651dbaf382392f60";
+
+    anaWallet = createFromPrivateKey(anaPrivateKey);
+
+    // Contract deployment setup with ana (accounts[0]) as the only owner
+    // To change that, edit the file "tasit-contract/3rd-parties/gnosis/scripts/2_deploy_contracts.js"
+    gnosisSafeContract = new GnosisSafe(GNOSIS_SAFE_ADDRESS);
+
+    erc20 = new DetailedERC20(ERC20_ADDRESS);
+  });
+
+  beforeEach("", async () => {
+    snapshotId = await createSnapshot(provider);
+
+    const etherBalance = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${etherBalance}`).to.equal(`${ZERO}`);
+
+    const erc20Balance = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${erc20Balance}`).to.equal(`${ZERO}`);
+
+    //gnosisSafeContract.removeWallet();
+    //erc20.removeWallet();
+  });
+
+  afterEach("", async () => {
+    await revertFromSnapshot(provider, snapshotId);
+  });
+
+  it("wallet owner should deposit and withdraw some ethers", async () => {
+    anaWallet = anaWallet.connect(provider);
+    const tx = await anaWallet.sendTransaction({
+      to: GNOSIS_SAFE_ADDRESS,
+      value: ONE,
+    });
+    await provider.waitForTransaction(tx.hash);
+
+    const balanceAfterDeposit = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterDeposit}`).to.equal(`${ONE}`);
+
+    const signers = [anaWallet];
+    const toAddress = anaWallet.address;
+    const value = ONE;
+
+    gnosisSafeContract.setWallet(anaWallet);
+    const execTxAction = await gnosisSafeContract.sendEtherTransaction(
+      signers,
+      toAddress,
+      value
+    );
+    await execTxAction.waitForNonceToUpdate();
+
+    const balanceAfterWithdraw = await provider.getBalance(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
+  });
+
+  it("wallet owner should deposit and withdraw some ERC20 tokens", async () => {
+    erc20.setWallet(anaWallet);
+    const mintAction = erc20.mint(anaWallet.address, ONE);
+    await mintAction.waitForNonceToUpdate();
+
+    const transferAction = erc20.transfer(GNOSIS_SAFE_ADDRESS, ONE);
+    await transferAction.waitForNonceToUpdate();
+
+    const balanceAfterDeposit = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterDeposit}`).to.equal(`${ONE}`);
+
+    const signers = [anaWallet];
+    const tokenAddress = erc20.getAddress();
+    const toAddress = anaWallet.address;
+    const value = ONE;
+
+    gnosisSafeContract.setWallet(anaWallet);
+    const execTxAction = await gnosisSafeContract.sendERC20Transaction(
+      signers,
+      tokenAddress,
+      toAddress,
+      value
+    );
+    await execTxAction.waitForNonceToUpdate();
+
+    const balanceAfterWithdraw = await erc20.balanceOf(GNOSIS_SAFE_ADDRESS);
+    expect(`${balanceAfterWithdraw}`).to.equal(`${ZERO}`);
```

Comment:
Maybe also `expect` that it's back to `${ONE}` for Ana's normal account?

Reply:
OK, sure.

Reply:
Extracted to: https://github.com/tasitlabs/TasitSDK/issues/237

Reply:
Thanks. Resolving